docs/v3/tutorials/airflow.mdx

```diff
@@ -0,0 +1,502 @@
+
+Prefect's approach **removes unnecessary abstraction layers**, allowing direct access to the full Python ecosystem without Airflow-specific constraints.
+
+Basically: **anything done with a custom Airflow operator or hook can be replaced in Prefect with a task using the appropriate Python library.** Prefect removes Airflow's constraints, allowing direct use of the full Python ecosystem. For example, instead of using a **BashOperator** to call an API via a shell script, install the required package in your environment and call it directly from a task, eliminating unnecessary workarounds.
+
+### Observability
+
+#### State and logging
+**Task and Flow State Management**
+In Airflow, task states (`success`, `failed`, `skipped`, etc.) are stored in a metadata database and displayed in the Airflow UI's DAG run view. Prefect also tracks state for **each task and flow run**, but these states are managed by the **Prefect backend** (Prefect Server or Cloud API) and can be accessed via the **Prefect UI, API, or CLI**.
+
+After migration, similar visibility is available in Prefect's UI, where you can track which flows and tasks succeeded or failed. Prefect also includes additional state management features such as: 
+- **Flow cancellation** (`Cancelling` state).
+- **Resuming failed flows** (with manual steps).
+- **Task caching** between runs to avoid redundant computations.
+
+**Logging Differences**
+Airflow logs task execution output to files (stored on executor machines or remote storage), viewable through the UI. Prefect **captures stdout, stderr, and Python logging** from tasks and sends them to the Prefect backend, making logs accessible in the **Prefect UI, API, and CLI**.
+
+To ensure logs appear correctly in Prefect's UI, use:
+```python
+@flow(log_prints=True)
+@task(log_prints=True)
+```
+These flags route `print()` statements to Prefect logs automatically.
+
+For centralized logging (e.g., ElasticSearch, Stackdriver), Prefect supports **custom logging handlers** and **third-party integrations**. Logs can be forwarded similarly to how Airflow handled external logging.
+
+**Debugging and Troubleshooting**
+Prefect simplifies debugging because tasks are **standard Python functions**. Instead of analyzing scheduler or worker logs, you can:
+- **Re-run individual tasks or flows locally** to reproduce issues.
+- **Test flows interactively** in an IDE before deploying.
+
+This direct execution model eliminates the need to troubleshoot failures through a scheduling system, making debugging faster and more intuitive than in Airflow.
+
+#### Monitoring
+**Notifications and Alerts**
+In Airflow, monitoring is typically managed through the UI, email alerts on task failures, and external monitoring of the scheduler. Prefect provides similar capabilities through **Notifications and Automations**, which can be configured to trigger alerts via **Slack, email, or webhooks** based on specific events.
```

Comment:
Another nit - I think notifications are deprecated so maybe better to just talk about automations here.